src/MLJGLMInterface.jl

```diff
@@ -168,19 +190,42 @@ glm formula and glm data header.
 """
 function glm_features(model, X)
     if Tables.columnaccess(X)
-        table_features = keys(Tables.columns(X))
+        table_features = _to_array(keys(Tables.columns(X)))
     else
         first_row = iterate(Tables.rows(X), 1)[1]
-        table_features = first_row === nothing ? Symbol[] : keys(first_row)
+        table_features = first_row === nothing ? Symbol[] : _to_array(keys(first_row))
     end
-    features = filter(x -> x != model.offsetcol, table_features)
+    features = filter!(x -> x != model.offsetcol, table_features)
     return features
```

Comment:
```suggestion
    filter!(!=(model.offsetcol), table_features)
    return table_features
```
When using the mutation function, it's more clear to not rename the variable